### PR TITLE
Fix flaky test

### PR DESCRIPTION
**Pull request by GinnyJI**
From the fail logs on issue #871, the flaky tests all happen after a
HNCConfigue update, but the supposed follwing updates could not be
detected. We give these lines more time in Eventually() so that
HNCConfigue update gets more time to propogate. However, I was not able
to reproduce the flaky test results when running locally, so I cannot
confirm that they are fully fixed by this pr.

Tested: make test

### CODE REVIEW

File: incubator/hnc/internal/reconcilers/hnc_config_test.go [hunk 1]
```diff
@@ -138,12 +138,12 @@ var _ = Describe("HNCConfiguration", func() {
 		addToHNCConfig(ctx, "v1", "Secret", api.Remove)
 
 		Eventually(hasHNCConfigurationConditionWithMsg(ctx, api.MultipleConfigurationsForOneType,
-			fmt.Sprintf("APIVersion: v1, Kind: Secret, Mode: %s", api.Remove))).Should(BeTrue())
+			fmt.Sprintf("APIVersion: v1, Kind: Secret, Mode: %s", api.Remove)), statusUpdateTime).Should(BeTrue())
```
Comment (adrianludwin) [inline incubator/hnc/internal/reconcilers/hnc_config_test.go:141]: Can you please update the comment for `statusUpdateTime` to include a reference to #871 ?

Comment (GinnyJI) [reply incubator/hnc/internal/reconcilers/hnc_config_test.go:142]: done

File: incubator/hnc/internal/reconcilers/hnc_config_test.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ const (
 	// experiments, tests are flaky when setting the statusUpdateTime to 3 seconds and tests can
 	// always pass when setting the time to 4 seconds. We may need to increase the time in future if
 	// the config reconciler takes longer to update the status.
+	// This issue is logged at https://github.com/kuhttps://github.com/kubernetes-sigs/multi-tenancy/issues/871
```
Comment (yiqigao217) [inline incubator/hnc/internal/reconcilers/hnc_config_test.go:34]: The link is not well-formed.

Comment (GinnyJI) [reply incubator/hnc/internal/reconcilers/hnc_config_test.go]: fixed

**Pull request merged**